core/main/src/firebolt/handlers/privacy_rpc.rs

```diff
@@ -549,9 +553,22 @@ impl PrivacyImpl {
                     "PrivacySettingsStorageType::Cloud: Not Available",
                 )))
             }
-            // PrivacySettingsStorageType::Sync => Err(jsonrpsee::core::Error::Custom(String::from(
-            //     "PrivacySettingsStorageType::Sync: Unimplemented",
-            // ))),
+            PrivacySettingsStorageType::Sync => {
+                let dist_session = platform_state.session_state.get_account_session().unwrap();
```

Comment:
Is it safe to unwrap() ?  Has the dist_session been validated ? 
In Ripple 1.0, we have a function named is_valid() which  checks if the distributor session has all the fields populated with valid values.

Reply:
Updated the pr to check if account session and privacy setting is Some then go on create a request param base on these structs.
In Ripple2.0, all the fields in account session are required so we just need to handle if account session is None.